packages/flutter/lib/src/widgets/router.dart

```diff
@@ -73,6 +73,35 @@ class RouteInformation {
   final Object? state;
 }
 
+/// A convenient bundle to configure a [Router] widget.
+///
+/// To configure a [Router] widget, one needs to provide several delegates,
+/// [RouteInformationProvider], [RouteInformationParser], [RouterDelegate],
+/// and [BackButtonDispatcher]. This abstract class provides way to bundle these
+/// delegates into a single object to configure a [Router].
+///
+/// The subclasses need to override each getter to return the corresponding
+/// delegate.
+///
+/// The [routerDelegate] must not be null. The [backButtonDispatcher],
+/// [routerInformationProvider], and [routerInformationProvider] are optional.
+///
+/// If [routerInformationProvider] is not null, [routeInformationParser] must
+/// also not be null.
+abstract class RouterConfig<T> {
```

Comment:
Could/should this be a concrete data class with a constructor like this:

```dart
RouterConfig<T>({required this.routerDelegate, this.routeInformationParser, ...})
```

That way, you can easily throw one together out of existing components.

third-party libraries could still implement it as an interface,